cpp/src/io/csv/reader_impl.cu

```diff
@@ -868,8 +868,8 @@ table_with_metadata read_csv(cudf::io::datasource* source,
         // quotechars in quoted fields results in reduction to a single quotechar
         // TODO: Would be much more efficient to perform this operation in-place
         // during the conversion stage
-        std::string const quotechar(1, parse_opts.quotechar);
-        std::string const dblquotechar(2, parse_opts.quotechar);
+        cudf::string_scalar const quotechar(std::string(1, parse_opts.quotechar), true, stream);
+        cudf::string_scalar const dblquotechar(std::string(2, parse_opts.quotechar), true, stream);
```

Comment:
I opened https://github.com/rapidsai/cudf/pull/14443 for this one, let me know how you want to proceed.

Reply:
https://github.com/rapidsai/cudf/pull/14443 looks good, and can be merged.

Reply:
I pulled the `string_scalar` out of the loop so we don't create them for each column.